### PR TITLE
[22.05] Fixed tool_utils types & schema-salad pin

### PR DESCRIPTION
**Pull request by mr-c**
Backport of https://github.com/galaxyproject/galaxy/pull/14368

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/tool_util/deps/containers.py [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+    Optional,
+    TYPE_CHECKING,
+)
 
 from galaxy.util import (
     asbool,
     plugin_config,
 )
 from .container_classes import (
+    Container,
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/deps/containers.py]:
```suggestion
```

Comment (mr-c) [reply lib/galaxy/tool_util/deps/containers.py]: Merci beaucoup from CDG!

File: packages/tool_util/setup.cfg [hunk 2]
```diff
@@ -55,6 +55,7 @@ console_scripts =
 [options.extras_require]
 cwl =
     cwltool==3.1.20211107152837
+    schema-salad==8.3.20220525163636
```
Comment (nsoranzo) [inline packages/tool_util/setup.cfg:58]: I think I missed why we need to pin schema-salad, isn't it pinned enough in cwltool? Currently the latter has `schema-salad>=8.2.20211104054942,<9` .

Comment (mr-c) [reply packages/tool_util/setup.cfg:58]: Anything after this pinned version breaks some tests. I can change it to an upper limit

Comment (nsoranzo) [reply packages/tool_util/setup.cfg:58]: Weird, I can replicate locally the failure of `test_boolean_defaults` unit test with schema-salad==8.3.20220721194857 if I run: `pytest test/unit/tool_util/test_cwl.py` but not if I execute it alone with `pytest test/unit/tool_util/test_cwl.py::test_boolean_defaults`
Not sure if that rings a bell?

I'd prefer not to have to pin schema-salad directly in the long run, if possible.

Comment (nsoranzo) [reply packages/tool_util/setup.cfg:58]: I've done some bisecting and this is the commit that introduces the issue: https://github.com/common-workflow-language/schema_salad/commit/4eda8147e9709a2c0f25da674d4b46f3283f40a6

Any idea @mr-c @jmchilton ?

**Pull request merged**